models/yang/sonic/sonic-route-map.yang

```diff
@@ -159,17 +159,34 @@ module sonic-route-map {
                 }
 
                 leaf-list match_neighbor {
-                    type inet:ip-address;
+                    type union {
+                        type inet:ip-address;
+                        type leafref {
+                            path "/prt:sonic-port/prt:PORT/prt:PORT_LIST/prt:ifname";
+                        }
+                        type leafref {
+                            path "/spc:sonic-portchannel/spc:PORTCHANNEL/spc:PORTCHANNEL_LIST/spc:name";
+                        }
+                        type leafref {
+                            path "/svlanint:sonic-vlan-interface/svlanint:VLAN_INTERFACE/svlanint:VLAN_INTERFACE_LIST/svlanint:vlanName";
+                        }
+                    }
                     description
-                        "List of IP addresses in the neighbor set";
-                    max-elements 1;
+                        "Match neighbor IP or interface.";
+                    must "(count(current()) = 1)" {
+                        error-app-tag "max-allowed-match-nbr-reached";
+                        error-message "Only one match neighbor can be configured.";
+                    }
                 }
 
                 leaf-list match_tag {
                     type tag-type;
                     description
                         "Value of the tag set member";
-                    max-elements 1;
+                    must "(count(current()) = 1)" {
```

Comment:
Why we have chosen leaf-list if it is going to contain only one element ? We should use leaf in that case.
This does not seem correct modelling to me.

Reply:
Partha - We have two options, 1) Dont deviate OC model but put the restriction based on backend capability 2) Deviate OC model based on backend capability, we chose the first one.

Reply:
OC model and SONiC YANG model can be different.

Reply:
Yes, but if we want to keep leaf in the SONIC YANG and leaf-list in OC model, we should make the changes in the transformer function to handle appropriately.

Reply:
Can we support more than one element in leaf-list in any platform in SONiC ?

If not, we should make it as leaf instead of leaf-list in SONiC YANG.

As you pointed out, we have to write transformer code for handling such conversion between OC and SONiC YANG model.

Reply:
yes, we do have leaf-list in sonic YANG to hold multiple values e.g set_community_inline, we can make the changes appropriately.